### ISSUE TITLE
Test Chaos Mesh on public cloud service

### ISSUE THREAD

**Issue by YangKeao** [chaos-mesh/chaos-mesh#1182]
There have been a lot of issues about running Chaos Mesh on a public cloud service such as GKE, AWS EKS, TKE... These managed Kubernetes services can bring a lot of problems to Chaos Mesh:

1. The CNI plugin may not support `hostPort`

2. The kernel may be pruned too much so that some required kernel modules/configs are not enabled.

In this issue, I will write down a list to track the process to test Chaos Mesh on the public cloud service. If a special configuration is needed, we should also write documents for them.

- [x] GKE
- [ ] AWS EKS
- [ ] Azure AKS
- [ ] Tencent Cloud TKE
- [ ] Aliyun ACK

Feel free to comment and enrich the list together 🍻 .

**Comment by Colstuwjx** [comment]
+ HuaweiCloud CCE.

**Comment by STRRL** [comment]
Refs to https://github.com/chaos-mesh/chaos-mesh/issues/1224, maybe we should test chaos-mesh on "VMware Tanzu Kubernetes Grid"

**Comment by YangKeao** [comment]
@fewdan How is it going?

**Comment by leosunmo** [comment]
Has anyone tried this on Azure AKS? We're not running any Windows VMs or containers, just the usual Ubuntu NodeGroups on AKS. Having some issues at the moment, and wondering if it's simply not supported since AKS has its own CNI by the looks of it.

**Comment by YangKeao** [comment]
@fewdan Do you have plan to test on Azure AKS? When will it be tested?

**Comment by benbp** [comment]
I've found a couple things testing Azure AKS with Chaos Mesh:

- Chaos Mesh works fine in my limited testing on stock AKS clusters (which use kubenet by default).
  - FYI clusters on Kubernetes version >=1.19 [use containerd](https://docs.microsoft.com/en-us/azure/aks/cluster-configuration?utm_source=thenewstack&utm_medium=website&utm_campaign=platform#container-runtime-configuration), so the appropriate chaosDaemon helm values need to be set, e.g.:
  ```
    helm install chaos-mesh/chaos-mesh --name=chaos-mesh --namespace=chaos-testing \
      --set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock
  ```
  - At least for network chaos, I found it works for both docker and containerd runtimes.
- Chaos Mesh has some issues with [Azure CNI](https://docs.microsoft.com/en-us/azure/aks/configure-azure-cni) enabled clusters (clusters deployed with `--network-plugin azure`).
  - The behavior I see is that just **installing the chaos helm chart breaks pod DNS**. Even after un-installing the chart, pod DNS remains broken. Both internal and external cluster DNS queries from the pod just time out (dig/nslookup/etc. just hang, basically). DNS via a `kubectl debug` shell on the node still works. Pod networking otherwise seems fine as I can access internal and external IPs.
  - I haven't found any changes to `/etc/resolv.conf`, core dns configmaps, etc. but I assume someone more familiar with the chaos-daemon may know what the issue could be.

**Comment by YangKeao** [comment]
@benbp Thanks for your testing.

> The behavior I see is that just **installing the chaos helm chart breaks pod DNS**

:fearful:  It's really amazing that it will break the DNS, as the chaos mesh nearly does nothing on creating (only creates a lot of pods/deployments/deamonsets/webhooks/...), and these components won't do anything while there is no chaos resource. 

I will try to investigate and fix these problems on AKS as soon as possible (as the problem seems severe, breaking users' cluster is really not expected).

**Comment by benbp** [comment]
@YangKeao yeah it doesn't make sense to me why it would either. I did more testing with a simpler Azure CNI-enabled cluster (nothing more than `az aks create -g <group> -n <name> --network-plugin azure`, kubernetes version 1.19.11). In this case, **chaos mesh DID work** just fine. Pod networking was functional, and network chaos experiments also worked when created/deleted. Sorry for raising alarms!

Now I'm wondering whether my previous deployments, which had some extra virtual network configurations, were what was causing the issue, and I was misled in thinking it was Azure CNI. I'll try to narrow down a minimal configuration that reproduces the behavior I was seeing.

**Comment by shukawam** [comment]
please add Oracle Cloud Infrastructure OKE to list :)

**Comment by STRRL** [comment]
Just curious, could Chaos Mesh work on GKE auto-pilot cluster?

**Comment by YangKeao** [comment]
> Just curious, could Chaos Mesh work on GKE auto-pilot cluster?

No

**Comment by joelcomp1** [comment]
verified on Oracle OKE that config is 

```yaml
chaosDaemon:
  runtime: crio
  socketPath: /var/run/crio/crio.sock
```
